apps/src/templates/manageStudents/manageStudentsRedux.js

```diff
@@ -832,3 +853,28 @@ const transferStudentsOnServer = (
       onComplete(status, jqXhr.responseJSON);
     });
 };
+
+export const sectionStudentsDataOnServer = sectionId => {
+  return (dispatch, getState) => {
+    const state = getState().manageStudents;
+    // Don't load data if it's already stored in redux.
+    const alreadyHaveStudentData =
+      Object.values(state.studentData)[0] &&
+      Object.values(state.studentData)[0].sectionId === sectionId;
```

Comment:
please check how this behaves when you have no students

Reply:
When you have no students alreadyHaveStudentData comes back undefined so we try to load again. I played with finding the number of students from sectionDataRedux but the number there does not stay in sync with the number in the table if you add a new student using the table.

Reply:
One idea would be to add a `sectionStudentsLoaded` array to redux, whose sole purpose is to record whether you've already loaded the students for this section, so that you can decide here whether to request them again.